src/pages/register-a-beacon/check-beacon-details.tsx

```diff
@@ -15,85 +15,144 @@ import {
 import { Details } from "../../components/Details";
 import { IfYouNeedHelp } from "../../components/Mca";
 import { GetServerSideProps, GetServerSidePropsContext } from "next";
-import { BeaconCacheEntry } from "../../lib/form-cache";
+import { BeaconCacheEntry } from "../../lib/formCache";
 import { updateFormCache } from "../../lib/middleware";
 import { ErrorSummary } from "../../components/ErrorSummary";
+import { FieldValidator } from "../../lib/fieldValidator";
 
 interface CheckBeaconDetailsProps {
-  manufacturerError?: boolean;
-  modelError?: boolean;
-  hexIdError?: boolean;
+  manufacturer: string;
+  model: string;
+  hexId: string;
+  needsValidation: boolean;
 }
 
-interface BeaconManufacturerInputProps {
-  isError: boolean;
-}
-
-interface BeaconModelInputProps {
-  isError: boolean;
-}
-
-interface BeaconHexIdSelectProps {
+interface FormInputProps {
+  value: string;
   isError: boolean;
+  errorMessages: Array<string>;
 }
 
 interface ErrorMessageProps {
   id: string;
   message: string;
 }
 
+const manufacturerField = new FieldValidator("manufacturer");
+
+manufacturerField
+  .should()
+  .containANonEmptyString()
+  .withErrorMessage("Manufacturer should not be empty");
+
+const modelField = new FieldValidator("model");
+
+modelField
+  .should()
+  .containANonEmptyString()
+  .withErrorMessage("Model should not be empty");
+
+const hexIdField = new FieldValidator("hexId");
+
+hexIdField
+  .should()
+  .containANonEmptyString()
+  .withErrorMessage("HEX ID should not be empty");
+
+hexIdField
+  .should()
+  .beExactly15Characters()
+  .withErrorMessage("Hex ID should be 15 characters long");
+
 const CheckBeaconDetails: FunctionComponent<CheckBeaconDetailsProps> = ({
-  manufacturerError = false,
-  modelError = false,
-  hexIdError = false,
-}: CheckBeaconDetailsProps): JSX.Element => (
+  manufacturer,
+  model,
+  hexId,
+  needsValidation = false,
+}: CheckBeaconDetailsProps): JSX.Element => {
+  manufacturerField.value = manufacturer;
+  modelField.value = model;
+  hexIdField.value = hexId;
+
+  return (
+    <>
+      <Layout navigation={<BackButton href="/" />}>
+        <Grid
+          mainContent={
+            <>
+              {needsValidation &&
+                (manufacturerField.hasError() ||
+                  modelField.hasError() ||
+                  hexIdField.hasError()) && (
+                  <ErrorSummaryComponent
+                    validators={[manufacturerField, modelField, hexIdField]}
+                  />
+                )}
```

Comment:
`<ErrorSummary>` now only displays if `needsValidation` is true (i.e. the request that prompted the page to be rendered was a `POST` request)